### PR TITLE
Add artifacts to builds, add docker hub automatic releases, add githu…

### PR DESCRIPTION
**Pull request by evolvedlight**
This addresses #35 

It requires that the DOCKER_USERNAME and DOCKER_PASSWORD is set on your repository.

Example results: https://hub.docker.com/repository/docker/evolvedlight/kvrocks-exporter/general

https://github.com/evolvedlight/kvrocks_exporter/releases/tag/v1.0.17

In addition, each PR will create artifacts that could be tested.

I removed gox from the makefile as the upstream tool appears archived.
I also replaced the goreleaser multidocker build with kos as it seems easier to maintain.

### CODE REVIEW

**Comment by git-hulk** [comment]
@evolvedlight Thanks for your contributions.

File: Makefile [hunk 1]
```diff
@@ -63,19 +63,8 @@ upload-coverage:
...
 build-binaries:
-	go install github.com/oliver006/gox@master
-
-	rm -rf .build | true
+	go install github.com/goreleaser/goreleaser/v2@latest
 
-	export CGO_ENABLED=0 ; \
-	gox -os=$(OS) -arch=$(ARCH) -verbose -rebuild -ldflags $(GO_LDFLAGS) -output ".build/kvrocks_exporter" && \
-	echo "done"
+	goreleaser release --clean -f .github/goreleaser.yml
```
Comment (git-hulk) [inline Makefile:70]: Could we separate the `build` and `release` binary into two targets? That said, we can have `make build` to build to binary which doesn't depend on goreleaser, and add the target `make release` for releasing propose. What do you think about this?

Comment (evolvedlight) [reply Makefile:74]: Sure, made that change. My opinion I guess would be that this goreleaser step is never used and the release is only done by CI. Good then that the make build only makes a local build with no dependencies.

Comment (evolvedlight) [reply Makefile:74]: And thanks for checking the PR - I'd suggest that assuming it works, I'd make another PR to update the readme a bit to also point to the docker hub reference.

Comment (git-hulk) [reply Makefile:74]: Sure, thank you!

**Review by git-hulk — APPROVED** [review]

**Pull request merged**